src/test/java/javaserver/ResponseBuilderTest.java

```diff
@@ -60,30 +51,38 @@ public void testRespondsWith302() {
 
 	@Test
 	public void testFourOhFourResponseCode() throws IOException {
-		ResponseBuilder responder = new ResponseBuilder(unacceptableRequest);
-		Response response = responder.getResponse();
-		assertEquals(response.getResponseCode(), fourOhFour);
+		Response unsupportedRequestResponse = getResponse("GET", "/foo");
+		assertEquals(unsupportedRequestResponse.getResponseCode(), fourOhFour);
 	}
 
 	@Test
 	public void testMethodNotAllowedResponseCode() throws IOException {
-		ResponseBuilder responder = new ResponseBuilder(unallowedRequest);
-		Response response = responder.getResponse();
-		assertEquals(response.getResponseCode(), fourOhFive);
+		Response unallowedRequestResponse = getResponse("POST", "/file1");
+		assertEquals(unallowedRequestResponse.getResponseCode(), fourOhFive);
 	}
 
 	@Test
 	public void testOptionsHeader() throws IOException {
-		ResponseBuilder responder = new ResponseBuilder(requestWithOptions);
-		Response response = responder.getResponse();
-		assertEquals(response.getHeader(), methodOptionsHeader);
+		Response optionsResponse = getResponse("GET", "/method_options");
+		assertEquals(optionsResponse.getHeader(), methodOptionsHeader);
 	}
 
 	@Test
 	public void testRedirectHeader() throws IOException {
-		ResponseBuilder responder = new ResponseBuilder(redirectRequest);
-		Response response = responder.getResponse();
-		assertEquals(response.getHeader(), redirectHeader);
+		Response redirectResponse = getResponse("GET", "/redirect");
+		assertEquals(redirectResponse.getHeader(), redirectHeader);
+	}
+
+	@Test
+	public void testValidCodedParameters200() {
+		Response validParamsResponse = getResponse("GET", codedURI);
+		assertEquals(validParamsResponse.getResponseCode(), twoHundred);
+	}
+
+	@Test
+	public void testInvalidCodedParams404() {
+		Response invalidParamResponse = getResponse("GET", "/foo?");
+		assertEquals(invalidParamResponse.getResponseCode(), fourOhFour);
```

Comment:
I agree - that is too specific. Technically, any request can contain a query string with parameters and it's up to whatever route it hits to decide if they actually mean anything to it. A real world example is if you were to go to something like `google.com?not_a_parameter=hello`. The request is still valid, and you'll receive a 200 with a response, but the query string doesn't mean anything to this route so it's just ignored.